### PR TITLE
Remove overuse of Optional.

### PR DESCRIPTION
**Pull request by TonyWardell**

### CODE REVIEW

File: sources/src/main/java/uk/ac/ebi/quickgo/ff/loader/ontology/AbstractGenericOLoader.java [hunk 1]
```diff
@@ -19,216 +17,260 @@
...
 
-    public AbstractGenericOLoader(S sources) {
+    AbstractGenericOLoader(S sources) {
         super(sources);
     }
 
     protected abstract T newInstance();
 
-    protected T createWithGenericOInfo(String nameSpace, Optional<String> optionalRootTermId) throws Exception {
+    T createWithGenericOInfo(String nameSpace, String optionalRootTermId) throws Exception {
```
Comment (eddturner) [inline sources/src/main/java/uk/ac/ebi/quickgo/ff/loader/ontology/AbstractGenericOLoader.java]: The class "as-is" means it's okay for another class in the same package to create an `AbstractGenericOLoader` instance, then call `createWithGenericOInfo` -- but I think this method is only really applicable to concrete classes.

... I think changing the class to package-private, and retaining `createWithGenericOInfo` as protected hopefully makes this clearer.

File: indexing/src/main/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverter.java [hunk 2]
```diff
@@ -24,13 +23,11 @@
...
  * @author Edd
  */
-public class GenericTermToODocConverter implements Function<Optional<? extends GenericTerm>,
-        Optional<OntologyDocument>> {
+public class GenericTermToODocConverter implements Function<GenericTerm,
+        OntologyDocument> {
 
-    @Override public Optional<OntologyDocument> apply(Optional<? extends GenericTerm> termOptional) {
-        if (termOptional.isPresent()) {
+    @Override public OntologyDocument apply(GenericTerm term) {
```
[the remark this answers is not in the document]
Comment (TonyWardell) [reply indexing/src/main/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverter.java]: Done

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GOTermToODocConverterTest.java [hunk 3]
```diff
@@ -127,19 +126,18 @@ public void convertsSimpleFieldsWhenNotExists() {
...
         assertThat(document.aspect, is(nullValue()));
     }
 
     // empty optional conversion
-    @Test
-    public void convertsEmptyOptional() {
-        Optional<OntologyDocument> documentOptional = converter.apply(Optional.empty());
-        assertThat(documentOptional.isPresent(), is(false));
-    }
+//    @Test
```
[the remark this answers is not in the document]
Comment (TonyWardell) [reply indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GOTermToODocConverterTest.java]: Done

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/OntologyReaderTest.java [hunk 4]
```diff
@@ -132,7 +142,7 @@ public void opensSuccessfully() {
 
     @After
     public void after() {
-        ontologyReader.close();
+        if(ontologyReader != null) ontologyReader.close();
```
[the remark this answers is not in the document]
Comment (TonyWardell) [reply indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/OntologyReaderTest.java]: I have the checkbox selected, but there is a different behaviour if I select formatting manually cf to doing it via a commit.

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverterTest.java [hunk 5]
```diff
@@ -376,11 +370,11 @@ public void convertsSimpleNullFields() {
     }
 
     // empty optional conversion
-    @Test
-    public void convertsEmptyOptional() {
-        Optional<OntologyDocument> documentOptional = converter.apply(Optional.empty());
-        assertThat(documentOptional.isPresent(), is(false));
-    }
+//    @Test
```
[the remark this answers is not in the document]
Comment (TonyWardell) [reply indexing/src/test/java/uk/ac/ebi/quickgo/index/ontology/converter/GenericTermToODocConverterTest.java]: Done.
Added test for apply of null GoTerm

File: sources/src/main/java/uk/ac/ebi/quickgo/ff/loader/ontology/AbstractGenericOLoader.java [hunk 6]
```diff
@@ -19,216 +17,260 @@
 public abstract class AbstractGenericOLoader<S extends OntologySourceFiles, T extends GenericOntology>
         extends SourceInfoLoader<S, T> {
 
-    public AbstractGenericOLoader(S sources) {
+    AbstractGenericOLoader(S sources) {
         super(sources);
     }
 
     protected abstract T newInstance();
```
[the remark this answers is not in the document]
Comment (TonyWardell) [reply sources/src/main/java/uk/ac/ebi/quickgo/ff/loader/ontology/AbstractGenericOLoader.java]: Good spot removed.

File: sources/src/main/java/uk/ac/ebi/quickgo/ff/loader/ontology/AbstractGenericOLoader.java (hunk 1 above)
Comment (TonyWardell) [reply sources/src/main/java/uk/ac/ebi/quickgo/ff/loader/ontology/AbstractGenericOLoader.java]: Changed to protected.

File: sources/src/main/java/uk/ac/ebi/quickgo/ff/loader/ontology/OntologyGraphicsSourceLoader.java [hunk 7]
```diff
@@ -28,7 +28,7 @@
     private GeneOntology geneOntology;
     private EvidenceCodeOntology evidenceCodeOntology;
 
-    public OntologyGraphicsSourceLoader(File sourceDir) {
+    public OntologyGraphicsSourceLoader(File sourceDir){
```
[the remark this answers is not in the document]
Comment (TonyWardell) [reply sources/src/main/java/uk/ac/ebi/quickgo/ff/loader/ontology/OntologyGraphicsSourceLoader.java]: Reformat definitely not having an affect on commit.. manually reformatted.

File: sources/src/main/java/uk/ac/ebi/quickgo/ff/loader/ontology/OntologyGraphicsSourceLoader.java [hunk 8]
```diff
@@ -37,15 +37,10 @@ public OntologyGraphicsSourceLoader(File sourceDir) {
...
-                .map(go -> this.geneOntology = go);
-
-        new ECOLoader(new ECOSourceFiles(requireNonNull(sourceDir)))
-                .load()
-                .map(eco -> evidenceCodeOntology = eco);
-
-        if (isLoaded()) {
+        try {
+            this.geneOntology = new GOLoader(new GOSourceFiles(requireNonNull(sourceDir))).load();
+            evidenceCodeOntology = new ECOLoader(new ECOSourceFiles(requireNonNull(sourceDir))).load();
```
[the remark this answers is not in the document]
Comment (TonyWardell) [reply sources/src/main/java/uk/ac/ebi/quickgo/ff/loader/ontology/OntologyGraphicsSourceLoader.java]: Done.
Add try to load each ontology seperately as this is what the originally code was doing

File: sources/src/test/java/uk/ac/ebi/quickgo/ff/loader/ontology/GOLoaderIT.java [hunk 9]
```diff
@@ -34,11 +36,20 @@
...
+            geneOntology = goLoader.load();
+            assertThat(geneOntology, is(notNullValue()));
+
+            //Highly input data specific
+            assertEquals(((GOTerm)geneOntology.getTerm("GO:0003774")).getBlacklist().get(0).getMethodId(),
+                         "IPR001609|IPR002928");
+        } catch (Exception e) {
+            assertThat(e, is(nullValue()));
+        }
+
```
[the remark this answers is not in the document]
Comment (TonyWardell) [reply sources/src/test/java/uk/ac/ebi/quickgo/ff/loader/ontology/GOLoaderIT.java]: Done,

[the remark this answers is not in the document]
Comment (TonyWardell) [reply sources/src/test/java/uk/ac/ebi/quickgo/ff/loader/ontology/GOLoaderIT.java]: Done.

**Comment by TonyWardell** [comment]
I believe I know what went on with the formatting, I believe I had applied your version of the formatting xml to a previous version of intellij (rc for 2017.3 ) which I then stopped using, went back to 2017.2  which didin't then have the latest version of uniprot.xml. What a tangled web we (I) weave.

**Pull request merged**